lib/screens/login_screen.dart

```diff
@@ -138,11 +172,20 @@ class _LoginScreenState extends State<LoginScreen> {
               height: 25.0,
             ),
             TextButton(
-              onPressed: () {
+              onPressed: () async {
                 setState(() {
                   _validate = _text.text.isEmpty;
                   _emailvalidate = _emailtext.text.isEmpty;
                 });
+
+                res = await ref
+                    .read(loginRepositoryProvider)
+                    .login(_text.text, _emailtext.text);
+
+                if (res.statusCode == 200) {
+                  Navigator.of(context).pushNamed('DocumentPage');
+                  print("Yes!!");
+                }
```

Comment:
You don't want this type of logic in your screen code. You should leverage something like `FutureProvider` to help with this and remove some of the boilerplate code. 

https://pub.dev/documentation/riverpod/latest/riverpod/FutureProvider-class.html